devenv/lib/repository.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import os.path
+from configparser import ConfigParser
+
+from devenv.lib.config import get_config
+from devenv.lib.fs import ensure_binroot
+
+
+class Repository:
+    path: str
+    name: str
+    config_path: str
+
+    def __init__(self, root: str) -> None:
+        self.path = root
+        self.name = root.split("/")[-1]
+
+        # .devenv may exist but not be config_path because .devenv/bin is hard-coded, so we check for config.ini
+        if os.path.exists(f"{root}/.devenv/config.ini"):
+            self.config_path = f"{root}/.devenv"
+        # devenv _may_ exist but not be config_path because devenv has a devenv module, so we check for config.ini
```

Comment:
I think this `if` block can be left off entirely -- it'll be handled by `else` in every case I can think of -- except maybe a symlink where *both* exist 🤔

Reply:
This is here for migration purposes. When we move the sync.py from devenv to .devenv a config.ini may have already been created. As the config wouldn't be checked into git, it'd remain until manually deleted by the user.